python/dolfin/function/expression.py

```diff
@@ -112,7 +113,21 @@ def __init__(self,
         shape: tuple
             Value shape.
         """
-        # Without this, undefined behaviour might happen, see pybind11 docs.
-        super().__init__(shape)
+        self._cpp_object = cpp.function.Expression(shape)
+
         self.shape = shape
-        self.set_eval(eval_func)
+
+        # If user provided a Numba C callback then take address and
+        # set to the underlying cpp class
+        if eval_func is not None:
+            self._eval_address = eval_func.address
+            self._cpp_object.set_eval(self._eval_address)
+
+    @property
+    def eval_address(self): # noqa
+        return self._eval_address
+
+    @eval_address.setter
+    def eval_address(self, value: int):
```

Comment:
Can we do everything via the initialiser and drop this?

Reply:
I wanted to avoid isinstance check in __init__

Reply:
Use `try`?